src/main/webapp/app/overview/course-lectures/video-unit/video-unit.component.ts

```diff
@@ -37,5 +44,19 @@ export class VideoUnitComponent implements OnInit {
     handleCollapse(event: Event) {
         event.stopPropagation();
         this.isCollapsed = !this.isCollapsed;
+
+        if (!this.isCollapsed) {
+            // Mark the unit as completed when the user has it open for at least 5 minutes
+            this.completionTimeout = setTimeout(() => {
+                this.onCompletion.emit({ lectureUnit: this.videoUnit, completed: true });
+            }, 1000 * 60 * 5);
```

Comment:
Can you also access the length of the video? Some courses use many small video units instead and a video might be shorter than five minutes.
I'd suggest using the minimum of the video length (or even less to account for 2x speed) and 5 minutes

Reply:
Unfortunately not, as they are embedded in an iFrame where Artemis doesn't have any access (we allow many different video sources, so we can not simply, e.g., use the [YouTube API](https://developers.google.com/youtube/iframe_api_reference)). Students can always manually mark the video as completed by just clicking on the checkmark.